src/ua/data_types/variant.rs

```diff
@@ -35,7 +37,7 @@ impl Variant {
     ///
     /// Returns `None` when the variant is empty.
     #[must_use]
-    pub fn value_type(&self) -> Option<ValueType> {
+    pub fn value_type(&self) -> Option<Result<ValueType, UnsupportedValueType>> {
```

Comment:
I deliberately preserved the result type to avoid introducing subsequent changes. Should be revisited.